src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,277 @@
+            System.out.println("\t\t" + task);
+            System.out.println(spacer + "Now you have " + taskList.size() + " tasks in the list.");
+        } catch (IllegalArgumentException e) {
+            System.err.println(spacer + e.getMessage());
+        }
+    }
+
+    /**
+     * Create a task from user's input command.
+     * @param cmd user command.
+     * @return a task instance (can be event, deadline, etc).
+     * @throws IllegalArgumentException when user input an invalid command
+     */
+    private static Task createTask(String cmd) throws IllegalArgumentException {
+        String[] cmdArgs = cmd.split(" ");
+        if (cmdArgs.length <= 1) {
+            throw new IllegalArgumentException("☹ OOPS!!! The description cannot be empty!");
+        }
+        switch (cmdArgs[0]) {
+        case "todo":
+            return new Todo(cmdArgs);
+        case "deadline":
+            return new Deadline(cmdArgs);
+        case "event":
+            return new Event(cmdArgs);
+        default:
+            throw new IllegalArgumentException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+        }
+    }
+
+    private static void initData(String path, String fileName) {
+        File directory = new File(path);
+        directory.mkdir();
+        File file = new File(path + "/" + fileName);
+        try {
+            file.createNewFile();
+        } catch (IOException e) {
+            System.out.println(e.getMessage());
+            e.printStackTrace();
+        }
+    }
+
+    private static void readData(String fileName) throws FileNotFoundException {
+        File file = new File(fileName);
+        if (!file.exists()) {
+            throw new FileNotFoundException();
+        }
+        Scanner fileScanner = new Scanner(file);
+        while (fileScanner.hasNext()) {
+            String line = fileScanner.nextLine();
+            String otherThanQuote = " [^\"] ";
+            String quotedString = String.format(" \" %s* \" ", otherThanQuote);
+            String regex = String.format("(?x) " + // enable comments, ignore white spaces
+                            ",                         " + // match a comma
+                            "(?=                       " + // start positive look ahead
+                            "  (?:                     " + //   start non-capturing group 1
+                            "    %s*                   " + //     match 'otherThanQuote' zero or more times
+                            "    %s                    " + //     match 'quotedString'
+                            "  )*                      " + //   end group 1 and repeat it zero or more times
+                            "  %s*                     " + //   match 'otherThanQuote'
+                            "  $                       " + // match the end of the string
+                            ")                         ", // stop positive look ahead
```

Comment:
This string literal seems to be too big, perhaps consider labelling it as a named constant so that reference to its context is easier to be understood by readers of your code.